src/routes/Profile/Profile.tsx

```diff
@@ -69,13 +85,10 @@ const Profile: React.FunctionComponent = () => {
         [variant]: 'claimed',
       }));
       setCollectionBadges((prev) => {
-        const copy = [...prev];
-        copy.forEach((b) => {
-          if (b.variant === variant) {
-            b.claimedAt = Date.now().valueOf();
-          }
-        });
-        return copy;
+        return prev.map((b) => ({
+          ...b,
+          claimedAt: b.variant === variant ? Date.now().valueOf() : b.claimedAt,
```

Comment:
trying to wrap my head around why claimedAt attribute is conditional on the variant

Reply:
not sure I understand the comment... keep in mind this code is to be removed as it mocks BE at the moment

Reply:
sure, no worries then.